packages/valora-auth/src/config.ts

```diff
@@ -0,0 +1,47 @@
+const functions = require('firebase-functions')
+// This allows us to access Cloud Function environment variables
+export const config = functions.config()
+
+export const MOONPAY_DATA = {
+  staging: {
+    url: config.moonpay.url_staging,
+    public_key: config.moonpay.public_key_staging,
+    private_key: config.moonpay.private_key_staging,
+  },
+  production: {
+    url: config.moonpay.url_prod,
+    public_key: config.moonpay.public_key_prod,
+    private_key: config.moonpay.private_key_prod,
+  },
+}
+
+export const TRANSAK_DATA = {
+  staging: {
+    url: config.transak.url_staging,
+    public_key: config.transak.public_key_staging,
+    private_key: config.transak.private_key_staging,
+  },
+  production: {
+    url: config.transak.url_prod,
+    public_key: config.transak.public_key_prod,
+    private_key: config.transak.private_key_prod,
+  },
+}
+
+export const RAMP_DATA = {
+  staging: {
+    url: config.ramp.url_staging,
+    public_key: config.ramp.public_key_staging,
+  },
+  production: {
+    url: config.ramp.url_prod,
+    public_key: config.ramp.public_key_prod,
+  },
+}
```

Comment:
I'm wondering about the need to handle staging/prod config at this level. i.e. within the same running instance.

Let's stay we deploy `valora-auth` on production (mainnet), do we need it to be able to serve staging stuff?
Similarly, we deploy it to staging (alfajores), it seems it shouldn't be able to serve urls that are meant for production.

There are 2 approaches:
- 1 deployment to serve a specific network (blockchain-api and notification-service do that now)
- 1 deployment to serve multiple networks (komenci, Odis)

Both approach work and have tradeoffs, I just want to be sure we make the right conscious decision about this.

Reply:
The reason I've put them into one deployment is because the code is exactly the same for staging and production. IMO splitting these out into separate staging and production deployments will lead to a lot of duplicate code and associated overhead with keeping the two deployments in sync.

A good point both you and Gonza raised is that I need to make it easier to deploy the same instance to a non-production environment so that we can make changes in staging without impacting production. I think this will give us the best of both worlds. What do you think?

Reply:
The reason I've put them into one deployment is because the code is exactly the same for staging and production. IMO splitting these out into separate staging and production deployments will lead to a lot of duplicate code and associated overhead with keeping the two deployments in sync.

A good point both you and Gonza raised is that I need to make it easier to deploy the same instance to a non-production environment so that we can make changes in staging without impacting production. I think this will give us the best of both worlds. What do you think?

Reply:
I've gone ahead with approach 2 - same deployment to multiple environments. I don't anticipate much deviation between staging and production (i.e., I don't think we'll ever want the staging environment to be different from the production environment for a prolonged period of time) so I thought this approach was cleanest

Reply:
A few more considerations:
- there's also a security implication to this, ideally we don't want production config (private keys) being deployed to a staging environment. i.e. staging and prod are better fully separated.
- I'm not sure where you see more code duplication, it's only the config that should change between staging and prod.
- lastly @gnardini mentioned an evolution of this service where it will need to write to the RTDB, so that transaction hashes from these providers can be correctly mapped in the user transaction feed. Right now `blockchain-api` and `notification-service` can do this easily without additional setup as they are deployed in the same GCP project as the Firebase RTDB. Doing this with approach 2 is more setup and more risk of writing to the wrong RTDB. 

Looks like we could have a sync chat about this.

Reply:
Just a note on that last point, the service that would write to the RTDM is the `notification-service`. How it will work is that we receive a webhook from providers, see if the tx hash belongs to a Valora address and write to the DB if so. The part we would need to change on this service is only the URL of Ramp to add a `webhookStatusUrl` param (the other providers configure webhooks in a different way).

Reply:
Hum I would argue that `notification-service` should only worry about push notifications.
It would be better to group webhook handling for CICO in the same place where we set the webhook.

Reply:
I can see the reasoning for that, but lets discuss this sync :)

Reply:
@jeanregisser I've split out the staging and production key configs. I think I misunderstood what you were saying but agree this is better.

I don't view our webhook server to be an evolution of this service. The scope of this service/PR is improved API key management and I envision that will extend to non-CICO providers as well (e.g., API keys for Google reCaptcha). How best to architect this is certainly open for discussion but I'd rather have it separate from this PR.

I believe I've adequately addressed your concerns with my latest commit but if you don't feel like that's the case, please feel free to schedule some time for us to sync on this :)